LambdaRuntimeDockerfiles/get-latest-aspnet-versions.ps1

```diff
@@ -0,0 +1,147 @@
+            
+            foreach ($ver in $previewVersions) {
+                if ($ver -match '(\d+)\.(\d+)\.(\d+)-preview\.(\d+)') {
+                    $major = [int]$matches[1]
+                    $minor = [int]$matches[2]
+                    $patch = [int]$matches[3]
+                    $preview = [int]$matches[4]
+                    
+                    $previewObj = New-Object PSObject
+                    Add-Member -InputObject $previewObj -MemberType NoteProperty -Name "OriginalVersion" -Value $ver
+                    Add-Member -InputObject $previewObj -MemberType NoteProperty -Name "Major" -Value $major
+                    Add-Member -InputObject $previewObj -MemberType NoteProperty -Name "Minor" -Value $minor
+                    Add-Member -InputObject $previewObj -MemberType NoteProperty -Name "Patch" -Value $patch
+                    Add-Member -InputObject $previewObj -MemberType NoteProperty -Name "Preview" -Value $preview
+                    
+                    $previewObjs += $previewObj
+                }
+            }
+            
+            # Sort by version components
+            $sortedPreviews = $previewObjs | Sort-Object -Property Major, Minor, Patch, Preview -Descending
+            
+            if ($sortedPreviews.Count -gt 0) {
+                $latestVersion = $sortedPreviews[0].OriginalVersion
+            } else {
+                # Fallback - just take the last one alphabetically
+                $latestVersion = ($previewVersions | Sort-Object)[-1]
+            }
+        }
+        else {
+            $latestVersion = $null
+        }
+        
+        if ($latestVersion) {
+            Write-Host "Latest ASP.NET Core runtime version for .NET $majorVersion is $latestVersion"
+            return $latestVersion
+        } else {
+            Write-Error "Could not determine latest version for .NET $majorVersion"
+            return $null
+        }
+    }
+    catch {
+        $errorMessage = "Error fetching versions for .NET $majorVersion " + $_
+        Write-Error $errorMessage
+        return $null
+    }
+}
+
+# Get latest versions for each .NET version
+$net8Version = Get-LatestAspNetVersion -majorVersion "8"
+$net9Version = Get-LatestAspNetVersion -majorVersion "9"
+$net10Version = Get-LatestAspNetVersion -majorVersion "10"
+
+# Verify we got valid versions
+$allVersionsValid = $true
+if (-not $net8Version) { 
+    Write-Error "Failed to determine .NET 8 version"
+    $allVersionsValid = $false
+}
+if (-not $net9Version) { 
+    Write-Error "Failed to determine .NET 9 version" 
+    $allVersionsValid = $false
+}
+if (-not $net10Version) { 
+    Write-Error "Failed to determine .NET 10 version" 
+    $allVersionsValid = $false
+}
+
+if (-not $allVersionsValid) {
+    exit 1
+}
+
+# Output as GitHub Actions environment variables
+Write-Output "NET_8_NEXT_VERSION=$net8Version"
+Write-Output "NET_9_NEXT_VERSION=$net9Version"
+Write-Output "NET_10_NEXT_VERSION=$net10Version"
```

Comment:
Instead of you outputting the versions in this ps1, let's call it from the github workflow in each .NET version's step. That way, this ps1 stays generic and wouldn't need updating everytime we add or remove a .NET version.

Reply:
fixed